src/event-store/services/event-store.service.interface.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  EventStoreProjection,
+  IPersistentSubscriptionConfig,
+} from '../../interfaces';
+import {
+  CreateContinuousProjectionOptions,
+  CreateOneTimeProjectionOptions,
+  CreateTransientProjectionOptions,
+  GetProjectionStateOptions,
+} from '@eventstore/db-client/dist/projections';
+import { DeletePersistentSubscriptionOptions } from '@eventstore/db-client/dist/persistentSubscription';
+import { PersistentSubscriptionSettings } from '@eventstore/db-client/dist/utils';
+import {
+  AppendResult,
+  BaseOptions,
+  Credentials,
+  StreamingRead,
+} from '@eventstore/db-client/dist/types';
+import {
+  AppendToStreamOptions,
+  GetStreamMetadataResult,
+  ReadStreamOptions,
+  SetStreamMetadataOptions,
+} from '@eventstore/db-client/dist/streams';
+import { StreamMetadata } from '@eventstore/db-client/dist/utils/streamMetadata';
+import { ReadableOptions } from 'stream';
+import { PersistentSubscription, ResolvedEvent } from '@eventstore/db-client';
+import { EventData } from '@eventstore/db-client/dist/types/events';
+
+export const EVENT_STORE_SERVICE = Symbol();
+
+export interface IEventStoreService {
+  createProjection(
+    query: string,
+    type: 'oneTime' | 'continuous' | 'transient',
+    projectionName?: string,
+    options?:
+      | CreateContinuousProjectionOptions
+      | CreateTransientProjectionOptions
+      | CreateOneTimeProjectionOptions,
+  ): Promise<void>;
+
+  getProjectionState<T>(
+    streamName: string,
+    options?: GetProjectionStateOptions,
+  ): Promise<T>;
+
+  updateProjections(projections: EventStoreProjection[]): Promise<void>;
+
+  assertProjections(projections: EventStoreProjection[]): Promise<void>;
```

Comment:
I'm wondering if "assert" is a consistent and user friendly keyword. We have createProjection, then for no reason assertProjections (does it mean upsert?).